pkg/webrtcclient/client.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webrtcclient
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pion/webrtc/v3"
+)
+
+type Signaling struct {
+	SendCh chan interface{}
+	RecvCh chan map[string]interface{}
+	// The ICE servers to use in the webRTC connection.
+	ICEServers []webrtc.ICEServer
+	// The servers that were created client side and need to be sent to the device.
+	// This is typically a subset of Servers. Ignored if empty.
+	ClientICEServers []webrtc.ICEServer
+}
+
+type Controller struct {
+	signaling      *Signaling
+	peerConnection *webrtc.PeerConnection
+	observer       Observer
+}
+
+func (c Controller) connect(onComplete func(error)) {
```

Comment:
is the value receiver intended here? If so, lets add a comment.

Reply:
nope, my mistake